product_code/user_service/impl/src/test/scala/de/upb/cs/uc4/user/impl/UserServiceSpec.scala

```diff
@@ -215,6 +217,28 @@ class UserServiceSpec extends AsyncWordSpec
         server.application.kafkaEncryptionUtility.decrypt[JsonUserData](container) should ===(JsonUserData(student0.username, Role.Student, forceDelete = true))
       }
 
+      "publish a soft-deleted user in minimal format" in {
+        val deletionSource: Source[EncryptionContainer, _] = client.userDeletionTopicMinimal().subscribe.atMostOnceSource
+        client.softDeleteUser(student0.username).handleRequestHeader(addAuthorizationHeader()).invoke()
+
+        val container: EncryptionContainer = deletionSource
+          .runWith(TestSink.probe[EncryptionContainer])
+          .request(1)
+          .expectNext(FiniteDuration(15, SECONDS))
+        server.application.kafkaEncryptionUtility.decrypt[JsonUsername](container) should ===(JsonUsername(student0.username))
+      }
+
+      "publish a soft-deleted user in precise format" in {
+        val deletionSource: Source[EncryptionContainer, _] = client.userDeletionTopicPrecise().subscribe.atMostOnceSource
+        client.softDeleteUser(student1.username).handleRequestHeader(addAuthorizationHeader()).invoke()
+
+        val container: EncryptionContainer = deletionSource
+          .runWith(TestSink.probe[EncryptionContainer])
+          .request(1)
+          .expectNext(FiniteDuration(15, SECONDS))
+        server.application.kafkaEncryptionUtility.decrypt[JsonUserData](container) should ===(JsonUserData(student0.username, Role.Student, forceDelete = true))
+      }
```

Comment:
Why does this test succeed? The call is soft-delete, but the test checks if the container has force-delete=true.

Reply:
It seems that the topic tests in general do not work as expected and need some rework

Reply:
We have fixed all topic tests.